### PR TITLE
Fix rowspan/colspan inside Table

### LINKED ISSUE #802: Table: colspan a rowspan attribute are not handled correctly
## Actual
image

## Expected
image

### PR DESCRIPTION
**Pull request by laurentC35**
This pr fix #802

### CODE REVIEW

File: src/components/lunatic-components.tsx [hunk 1]
```diff
@@ -1,26 +1,35 @@
...
+		| { label: ReactElement; colspan?: number; rowspan?: number }
+	)[];
 	// Key that trigger autofocus when it changes (pageTag)
 	autoFocusKey?: string;
 	// Returns the list of extra props to add to components
-	componentProps?: (component: FilledLunaticComponentProps) => T;
+	componentProps?: (
+		component: FilledLunaticComponentProps & {
+			colspan?: number;
+			rowspan?: number;
```
Comment (ddecrulle) [inline src/components/lunatic-components.tsx]: This seems to be a little bit hacky, I think correct way is to add `colspan` and `rowspan` props directly in `ComponentTableType`

File: src/components/lunatic-components.tsx [hunk 2]
```diff
@@ -1,26 +1,35 @@
...
+		}
+	) => T;
 	// Add additional wrapper around each component
 	wrapper?: (
 		props: PropsWithChildren<
-			FilledLunaticComponentProps & T & { index: number }
+			FilledLunaticComponentProps &
+				T & { index: number } & { colspan?: number; rowspan?: number }
```
Comment (ddecrulle) [inline src/components/lunatic-components.tsx]: Same

File: src/components/type.ts [hunk 3]
```diff
@@ -113,7 +113,10 @@ type ComponentPropsByType = {
 			rowspan?: number;
 			colspan?: number;
 		}>;
-		body: FilledLunaticComponentProps[][];
+		body: (FilledLunaticComponentProps & {
+			rowspan?: number;
+			colspan?: number;
+		})[][];
```
Comment (ddecrulle) [inline src/components/type.ts]: Same

File: src/use-lunatic/commons/fill-components/fill-specific-expression.ts [hunk 4]
```diff
@@ -124,7 +124,12 @@ function fillTable(
 				if (hasComponentType(component)) {
 					return fillComponent(component, state);
 				}
-				return state.executeExpression(getVTLCompatibleValue(component.label));
+				return {
+					...component,
+					label: state.executeExpression(
+						getVTLCompatibleValue(component.label)
+					),
+				};
```
Comment (ddecrulle) [inline src/use-lunatic/commons/fill-components/fill-specific-expression.ts]: Why do we need `...component` now ?

Comment (laurentC35) [reply src/use-lunatic/commons/fill-components/fill-specific-expression.ts]: To keep colspan and rowspan attribute which come from source.json (and lost if we don't keep other component's props)

File: src/components/lunatic-components.tsx (hunk 1 above)
Comment (laurentC35) [reply src/components/lunatic-components.tsx]: Yes, it will be more clear, I will correct this

**Comment by Grafikart** [comment]
Reprise via la #805

**Comment by laurentC35** [comment]
Close beacause of https://github.com/InseeFr/Lunatic/pull/805

**Pull request closed without merging**